Source/UI/CommonUI/Protocols/Elements/SideBorder.swift

```diff
@@ -0,0 +1,17 @@
+//
+//  SideBorder.swift
+//  Frames
+//
+//  Created by Ehab Alsharkawy on 26/10/2022.
+//  Copyright © 2022 Checkout. All rights reserved.
+//
+
+import UIKit
+
+public enum SideBorder: CaseIterable {
+    public static var allCases: [SideBorder] = [.left(layer: nil), .right(layer: nil), .top(layer: nil), .bottom(layer: nil)]
+    case left(layer: UIView?)
+    case right(layer: UIView?)
+    case top(layer: UIView?)
+    case bottom(layer: UIView?)
```

Comment:
We can make use of [Default enum associated values](https://www.swiftbysundell.com/tips/default-enum-associated-values/) which has been available since Swift 5.3 (so should have no impact on our minimum Swift 5.4 requirement).

Why UIView and not CALayer as originally proposed?